src/dust/coagulation/coagulation.cpp

```diff
@@ -0,0 +1,296 @@
+//========================================================================================
+// (C) (or copyright) 2023-2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+// Artemis includes
+#include "dust/coagulation/coagulation.hpp"
+#include "artemis.hpp"
+#include "dust/dust.hpp"
+#include "geometry/geometry.hpp"
+
+namespace Dust {
+namespace Coagulation {
+//----------------------------------------------------------------------------------------
+//! \fn  StateDescriptor Coagulalation::Initialize
+//! \brief Adds intialization function for coagulation package
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin, Params &dustPars) {
+  auto coag = std::make_shared<StateDescriptor>("coagulation");
+  Params &params = coag->AllParams();
+
+  CoagParams cpars;
+
+  const int nm = dustPars.template Get<int>("nspecies");
+  const Real dfloor = dustPars.template Get<Real>("dfloor");
+  const Real rho_p = dustPars.template Get<Real>("grain_density");
+  cpars.nm = nm;
+  cpars.vfrag = pin->GetOrAddReal("dust", "vfrag", 1.e3); // cm/s
+  cpars.nlim = 1e10;
+  cpars.integrator = pin->GetOrAddInteger("dust/coagulation", "coag_int", 3);
+  cpars.use_adaptive =
+      pin->GetOrAddBoolean("dust/coagulation", "coag_use_adaptiveStep", true);
+  cpars.mom_coag = pin->GetOrAddBoolean("dust/coagulation", "coag_mom_preserve", true);
+  cpars.nCall_mx = pin->GetOrAddInteger("dust/coagulation", "coag_nsteps_mx", 1000);
+  cpars.rho_p = rho_p;
+
+  const Real M_SUN = 1.988409870698051e33;      // gram (sun)
+  const Real GRAV_CONST = 6.674299999999999e-8; // gravitational const in cm^3 g^-1 s^-2
+  const Real mstar = pin->GetOrAddReal("problem", "mstar", 1.0) * M_SUN;
+  cpars.gm = std::sqrt(GRAV_CONST * mstar);
```

Comment:
```suggestion
  const Real mstar = pin->GetOrAddReal("problem", "mstar", 1.0) * constants.GetMsolarCode();
  cpars.gm = std::sqrt(constants.GetGCode() * mstar);
```
hardcoded constants need to be replaced with centralized constants from the `Constants` class introduced in #24